src/mpid/ch4/netmod/ofi/ofi_part.c

```diff
@@ -23,5 +23,12 @@ int MPIDI_OFI_mpi_precv_init_hook(void *buf, int partitions, MPI_Aint count,
 
 int MPIDI_OFI_precv_matched_hook(MPIR_Request * part_req)
 {
-    return MPI_SUCCESS;
+    int mpi_errno = MPI_SUCCESS;
+
+    /* If MPI_START has been called, notify sender CTS */
+    if (MPIR_Part_request_is_active(part_req)) {
+        mpi_errno = MPIDIG_part_issue_cts(part_req);
```

Comment:
We need clearly mark or separate the code that is supposed to be generic path only or common utility functions that is to be shared as part of common infrastructure. If you need this to be merged for now, I'll approve it. But I think I prefer you to make clean separations even that means some temporary duplication.